rpk: add `rpk debug remote-bundle`; collect a cluster-wide bundle

r-vasquez opened the pull request:
This PR adds the new command: `rpk debug remote-bundle` which lets the user collects a set of debug bundles from each node in the cluster. It uses the Admin API to do so. In order to collect the bundle, we have created 4 new commands:

- `rpk debug remote-bundle start`
- `rpk debug remote-bundle download`
- `rpk debug remote-bundle status`
-  `rpk debug remote-bundle cancel`

### Examples:

These are interactive-by-default commands, each interactive command has their respective `--no-confirm` flag to avoid confirmation prompts.

**rpk debug remote-bundle start** 
```
BROKER
127.0.0.1:11644
127.0.0.1:10644
127.0.0.1:9644
? Confirm debug bundle collection from these brokers? Yes
BROKER           JOB-ID
127.0.0.1:10644  7fe70e0d-472b-48a7-bc8a-046d04658074  
127.0.0.1:11644  7fe70e0d-472b-48a7-bc8a-046d04658074  
127.0.0.1:9644   7fe70e0d-472b-48a7-bc8a-046d04658074  

The debug bundle collection process has started with Job-ID 7fe70e0d-472b-48a7-bc8a-046d04658074, to check the 
status, run:
  rpk debug remote-bundle status
```
**rpk debug remote-bundle download** 
```
BROKER           STATUS   JOB-ID
127.0.0.1:9644   success  7fe70e0d-472b-48a7-bc8a-046d04658074
127.0.0.1:11644  success  7fe70e0d-472b-48a7-bc8a-046d04658074
127.0.0.1:10644  success  7fe70e0d-472b-48a7-bc8a-046d04658074
? Confirm debug bundle download from these brokers? Yes
BROKER           DOWNLOADED
127.0.0.1:9644   true  
127.0.0.1:10644  true  
127.0.0.1:11644  true  

Successfully downloaded remote debug bundle to 1730484634-remote-bundle.zip
```
**rpk debug remote-bundle status** 
```
BROKER           STATUS   JOB-ID
127.0.0.1:9644   running  7fe70e0d-472b-48a7-bc8a-046d04658074
127.0.0.1:11644  running  7fe70e0d-472b-48a7-bc8a-046d04658074
127.0.0.1:10644  running  7fe70e0d-472b-48a7-bc8a-046d04658074

After the process is completed, you may retrieve the debug bundle using:
  rpk debug remote-bundle download
```
**rpk debug remote-bundle cancel**
```
BROKER           STATUS   JOB-ID
127.0.0.1:10644  running  863d7e4f-f581-4f0e-9f3f-40b73dc7dcce
127.0.0.1:9644   running  863d7e4f-f581-4f0e-9f3f-40b73dc7dcce
127.0.0.1:11644  running  863d7e4f-f581-4f0e-9f3f-40b73dc7dcce
? Confirm debug bundle cancel from these brokers? Yes
BROKER           CANCELED
127.0.0.1:10644  true  
127.0.0.1:11644  true  
127.0.0.1:9644   true  

```

Additional work will be added in the future to collect everything on a single command, and also allow the user to clean-up the current cluster.

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v24.2.x
- [ ] v24.1.x
- [ ] v23.3.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

vbotbuildovich commented:
<!--NEW FAILURES-->
<!--743ebc41-aa8f-4bff-ad51-d4daccfb3d56-->
<!--body-->
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/57483#0192e917-5e91-479f-8c4c-ab61be9a455b:
```
"rptest.tests.control_character_flag_test.ControlCharacterPermittedAfterUpgrade.test_upgrade_from_pre_v23_2.initial_version=.22.2.9"
```
<!--/body-->

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/57483-->
<!--a173560a-c8a7-4f9f-af6b-31427546139f-->
<!--title-->
## Retry command for [Build#57483](https://buildkite.com/redpanda/redpanda/builds/57483)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/control_character_flag_test.py::ControlCharacterPermittedAfterUpgrade.test_upgrade_from_pre_v23_2@{"initial_version":[22,2,9]}
```
<!--/body-->

vbotbuildovich commented:
<!--RETRIED FAILURES-->
<!--397c85a0-d8cb-4a6a-b5da-add4fa0e541f-->
<!--body-->
ducktape was retried in https://buildkite.com/redpanda/redpanda/builds/57483#0192e917-5e8c-411e-bf42-1506a5294f90
<!--/body-->

Deflaimun commented on src/go/rpk/pkg/cli/debug/remotebundle/status.go:
```diff
@@ -0,0 +1,140 @@
+func newStatusCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "status",
+		Short: "Get the status of the current debug bundle process",
```
We should add long description just keep consistent to the others

Deflaimun commented on src/go/rpk/pkg/cli/debug/remotebundle/remote.go:
```diff
@@ -0,0 +1,62 @@
+func NewCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "remote-bundle",
+		Short: "Create a debug bundle from a remote cluster",
```
We should add long description

micheleRP commented on src/go/rpk/pkg/cli/debug/common/common.go:
```diff
@@ -0,0 +1,50 @@
+// InstallFlags installs the debug bundle flags that fills the debug bundle
+// options.
+func (o *DebugBundleSharedOptions) InstallFlags(f *pflag.FlagSet) {
+	f.StringVar(&o.ControllerLogsSizeLimit, "controller-logs-size-limit", "132MB", "The size limit of the controller logs that can be stored in the bundle (e.g. 3MB, 1GiB)")
```
```suggestion
	f.StringVar(&o.ControllerLogsSizeLimit, "controller-logs-size-limit", "132MB", "The size limit of the controller logs that can be stored in the bundle. For example: 3MB, 1GiB.")
```

r-vasquez replied:
Fixed. Except for adding the period as we don't add them to our flags help text which we found was a common pattern in other CLIs

micheleRP commented on src/go/rpk/pkg/cli/debug/common/common.go:
```diff
@@ -0,0 +1,50 @@
+// options.
+func (o *DebugBundleSharedOptions) InstallFlags(f *pflag.FlagSet) {
+	f.StringVar(&o.ControllerLogsSizeLimit, "controller-logs-size-limit", "132MB", "The size limit of the controller logs that can be stored in the bundle (e.g. 3MB, 1GiB)")
+	f.DurationVar(&o.CPUProfilerWait, "cpu-profiler-wait", 30*time.Second, "For how long to collect samples for the CPU profiler (e.g. 30s, 1.5m). Must be higher than 15s")
```
```suggestion
	f.DurationVar(&o.CPUProfilerWait, "cpu-profiler-wait", 30*time.Second, "How long to collect samples for the CPU profiler. For example: 30s, 1.5m. Must be higher than 15s.")
```

micheleRP commented on src/go/rpk/pkg/cli/debug/common/common.go:
```diff
@@ -0,0 +1,50 @@
+func (o *DebugBundleSharedOptions) InstallFlags(f *pflag.FlagSet) {
+	f.StringVar(&o.ControllerLogsSizeLimit, "controller-logs-size-limit", "132MB", "The size limit of the controller logs that can be stored in the bundle (e.g. 3MB, 1GiB)")
+	f.DurationVar(&o.CPUProfilerWait, "cpu-profiler-wait", 30*time.Second, "For how long to collect samples for the CPU profiler (e.g. 30s, 1.5m). Must be higher than 15s")
+	f.StringVar(&o.LogsSizeLimit, "logs-size-limit", "100MiB", "Read the logs until the given size is reached (e.g. 3MB, 1GiB)")
```
```suggestion
	f.StringVar(&o.LogsSizeLimit, "logs-size-limit", "100MiB", "Read the logs until the given size is reached. For example:  3MB, 1GiB.")
```

micheleRP commented on src/go/rpk/pkg/cli/debug/common/common.go:
```diff
@@ -0,0 +1,50 @@
+	f.StringVar(&o.ControllerLogsSizeLimit, "controller-logs-size-limit", "132MB", "The size limit of the controller logs that can be stored in the bundle (e.g. 3MB, 1GiB)")
+	f.DurationVar(&o.CPUProfilerWait, "cpu-profiler-wait", 30*time.Second, "For how long to collect samples for the CPU profiler (e.g. 30s, 1.5m). Must be higher than 15s")
+	f.StringVar(&o.LogsSizeLimit, "logs-size-limit", "100MiB", "Read the logs until the given size is reached (e.g. 3MB, 1GiB)")
+	f.StringVar(&o.LogsSince, "logs-since", "yesterday", "Include logs dated from specified date onward; (journalctl date format: YYYY-MM-DD, 'yesterday', or 'today'). Refer to journalctl documentation for more options")
```
```suggestion
	f.StringVar(&o.LogsSince, "logs-since", "yesterday", "Include logs dated from specified date onward. For example: journalctl date format: YYYY-MM-DD, 'yesterday', or 'today'. See the journalctl documentation for more options.")
```

r-vasquez replied:
What do you think of leaving just: `journalctl date format` as is the only option, and not an example of how to pass the flag?

micheleRP commented on src/go/rpk/pkg/cli/debug/common/common.go:
```diff
@@ -0,0 +1,50 @@
+	f.DurationVar(&o.CPUProfilerWait, "cpu-profiler-wait", 30*time.Second, "For how long to collect samples for the CPU profiler (e.g. 30s, 1.5m). Must be higher than 15s")
+	f.StringVar(&o.LogsSizeLimit, "logs-size-limit", "100MiB", "Read the logs until the given size is reached (e.g. 3MB, 1GiB)")
+	f.StringVar(&o.LogsSince, "logs-since", "yesterday", "Include logs dated from specified date onward; (journalctl date format: YYYY-MM-DD, 'yesterday', or 'today'). Refer to journalctl documentation for more options")
+	f.StringVar(&o.LogsUntil, "logs-until", "", "Include logs older than the specified date; (journalctl date format: YYYY-MM-DD, 'yesterday', or 'today'). Refer to journalctl documentation for more options")
```
```suggestion
	f.StringVar(&o.LogsUntil, "logs-until", "", "Include logs older than the specified date. For example: journalctl date format: YYYY-MM-DD, 'yesterday', or 'today'. See the journalctl documentation for more options.")
```

micheleRP commented on src/go/rpk/pkg/cli/debug/common/common.go:
```diff
@@ -0,0 +1,50 @@
+	f.StringVar(&o.LogsSizeLimit, "logs-size-limit", "100MiB", "Read the logs until the given size is reached (e.g. 3MB, 1GiB)")
+	f.StringVar(&o.LogsSince, "logs-since", "yesterday", "Include logs dated from specified date onward; (journalctl date format: YYYY-MM-DD, 'yesterday', or 'today'). Refer to journalctl documentation for more options")
+	f.StringVar(&o.LogsUntil, "logs-until", "", "Include logs older than the specified date; (journalctl date format: YYYY-MM-DD, 'yesterday', or 'today'). Refer to journalctl documentation for more options")
+	f.DurationVar(&o.MetricsInterval, "metrics-interval", 10*time.Second, "Interval between metrics snapshots (e.g. 30s, 1.5m)")
```
```suggestion
	f.DurationVar(&o.MetricsInterval, "metrics-interval", 10*time.Second, "Interval between metrics snapshots. For example: 30s, 1.5m.")
```

r-vasquez replied:
Changed.

One comment though on `e.g` vs `For example` that is worth having into consideration:

We attempt to keep the flags help text as short as possible as some commands are too cramped, that's why we used both `e.g` and `()` so it was 'easier' to spot the examples. take this one (`rpk debug bundle`) as an example:

Now:
![image](https://github.com/user-attachments/assets/e8835317-2f62-4749-9fae-8684fa086db3)

After the changes:
![image](https://github.com/user-attachments/assets/8fae28c2-d7a8-4d2e-8368-0235f5bf4b8d)

Deflaimun replied:
I think this is one of the cases where the CLI style-guide conflicts with docs style-guide. In this case, keep consistent to your own style-guide. cc @micheleRP

micheleRP commented on src/go/rpk/pkg/cli/debug/common/common.go:
```diff
@@ -0,0 +1,50 @@
+	f.StringVar(&o.LogsSince, "logs-since", "yesterday", "Include logs dated from specified date onward; (journalctl date format: YYYY-MM-DD, 'yesterday', or 'today'). Refer to journalctl documentation for more options")
+	f.StringVar(&o.LogsUntil, "logs-until", "", "Include logs older than the specified date; (journalctl date format: YYYY-MM-DD, 'yesterday', or 'today'). Refer to journalctl documentation for more options")
+	f.DurationVar(&o.MetricsInterval, "metrics-interval", 10*time.Second, "Interval between metrics snapshots (e.g. 30s, 1.5m)")
+	f.IntVar(&o.MetricsSampleCount, "metrics-samples", 2, "Number of metrics samples to take (at the interval of --metrics-interval). Must be >= 2")
```
```suggestion
	f.IntVar(&o.MetricsSampleCount, "metrics-samples", 2, "Number of metrics samples to take (at the interval of --metrics-interval). Must be >= 2.")
```

r-vasquez replied:
Same comment as the above on periods.

micheleRP commented on src/go/rpk/pkg/cli/debug/common/common.go:
```diff
@@ -0,0 +1,50 @@
+	f.StringVar(&o.LogsUntil, "logs-until", "", "Include logs older than the specified date; (journalctl date format: YYYY-MM-DD, 'yesterday', or 'today'). Refer to journalctl documentation for more options")
+	f.DurationVar(&o.MetricsInterval, "metrics-interval", 10*time.Second, "Interval between metrics snapshots (e.g. 30s, 1.5m)")
+	f.IntVar(&o.MetricsSampleCount, "metrics-samples", 2, "Number of metrics samples to take (at the interval of --metrics-interval). Must be >= 2")
+	f.StringArrayVarP(&o.PartitionFlag, "partition", "p", nil, "Comma-separated partition IDs; when provided, rpk saves extra admin API requests for those partitions. Check help for extended usage")
```
```suggestion
	f.StringArrayVarP(&o.PartitionFlag, "partition", "p", nil, "Comma-separated partition IDs. When provided, rpk saves extra Admin API requests for those partitions. See the help for extended usage.")
```

micheleRP commented on src/go/rpk/pkg/cli/debug/common/common.go:
```diff
@@ -0,0 +1,50 @@
+	f.DurationVar(&o.MetricsInterval, "metrics-interval", 10*time.Second, "Interval between metrics snapshots (e.g. 30s, 1.5m)")
+	f.IntVar(&o.MetricsSampleCount, "metrics-samples", 2, "Number of metrics samples to take (at the interval of --metrics-interval). Must be >= 2")
+	f.StringArrayVarP(&o.PartitionFlag, "partition", "p", nil, "Comma-separated partition IDs; when provided, rpk saves extra admin API requests for those partitions. Check help for extended usage")
+	f.StringVarP(&o.Namespace, "namespace", "n", "redpanda", "The namespace to use to collect the resources from (k8s only)")
```
```suggestion
	f.StringVarP(&o.Namespace, "namespace", "n", "redpanda", "The namespace to use to collect the resources from (K8s only).")
```

micheleRP commented on src/go/rpk/pkg/cli/debug/common/common.go:
```diff
@@ -0,0 +1,50 @@
+	f.IntVar(&o.MetricsSampleCount, "metrics-samples", 2, "Number of metrics samples to take (at the interval of --metrics-interval). Must be >= 2")
+	f.StringArrayVarP(&o.PartitionFlag, "partition", "p", nil, "Comma-separated partition IDs; when provided, rpk saves extra admin API requests for those partitions. Check help for extended usage")
+	f.StringVarP(&o.Namespace, "namespace", "n", "redpanda", "The namespace to use to collect the resources from (k8s only)")
+	f.StringArrayVarP(&o.LabelSelector, "label-selector", "l", []string{"app.kubernetes.io/name=redpanda"}, "Comma-separated label selectors to filter your resources. e.g: <label>=<value>,<label>=<value> (k8s only)")
```
```suggestion
	f.StringArrayVarP(&o.LabelSelector, "label-selector", "l", []string{"app.kubernetes.io/name=redpanda"}, "Comma-separated label selectors to filter your resources. For example: <label>=<value>,<label>=<value> (K8s only).")
```

micheleRP commented on src/go/rpk/pkg/cli/debug/remotebundle/cancel.go:
```diff
@@ -0,0 +1,121 @@
+		Short: "Cancel a remote bundle execution",
+		Long: `Cancel a remote bundle execution.
+
+This command will cancel the debug collection process in a remote cluster that
+you have configured either in your rpk profile, flags or environment variables.
+
+Alternatively, you may provide a --job-id to only cancel the debug bundle with
+the given Job ID.
+
+The flag '--no-confirm' can be used to avoid the confirmation prompt.
```
```suggestion
This command cancels the debug collection process in a remote cluster that
you configured in flags, environment variables, or your rpk profile.

Use the flag '--job-id' to only cancel the debug bundle with
the given job ID.

Use the flag '--no-confirm' to avoid the confirmation prompt.
```

micheleRP commented on src/go/rpk/pkg/cli/debug/remotebundle/download.go:
```diff
@@ -0,0 +1,216 @@
+		Short: "Download a remote debug bundle",
+		Long: `Download a remote debug bundle.
+
+This command will download the debug collection process in a remote cluster that
+you have configured either in your rpk profile, flags or environment variables.
+
+Alternatively, you may provide a --job-id to only download the debug bundle with
+the given Job ID.
+
+The flag '--no-confirm' can be used to avoid the confirmation prompt.
```
```suggestion
This command downloads the debug collection process in a remote cluster that
you configured in flags, environment variables, or your rpk profile.

Use the flag '--job-id' to only download the debug bundle with
the given job ID.

Use the flag '--no-confirm' to avoid the confirmation prompt.
```

micheleRP commented on src/go/rpk/pkg/cli/debug/remotebundle/start.go:
```diff
@@ -0,0 +1,217 @@
+		Short: "Start a remote debug bundle collection in your cluster",
+		Long: `Start a remote debug bundle collection in your cluster.
+
+This command will start the debug collection process in a remote cluster that
+you have configured either in your rpk profile, flags or environment variables.
+
+After starting the debug collection process, you may query the status with
+'rpk debug remote-bundle status' and download when is ready with
+'rpk debug remote-bundle download'.
```
```suggestion
This command starts the debug collection process in a remote cluster that
you configured in flags, environment variables, or your rpk profile.

After starting the debug collection process, you can query the status with
'rpk debug remote-bundle status', and when it's ready, you can download it with
'rpk debug remote-bundle download'.
```

micheleRP commented on src/go/rpk/pkg/cli/debug/remotebundle/start.go:
```diff
@@ -0,0 +1,217 @@
+'rpk debug remote-bundle status' and download when is ready with
+'rpk debug remote-bundle download'.
+
+The flag '--no-confirm' can be used to avoid the confirmation prompt.
```
```suggestion
Use the flag '--no-confirm' to avoid the confirmation prompt.
```

micheleRP commented on src/go/rpk/pkg/cli/debug/remotebundle/start.go:
```diff
@@ -0,0 +1,217 @@
+
+			if alreadyRunning {
+				fmt.Printf(`
+A debug bundle collection is already in process, you may cancel it by running:
```
```suggestion
A debug bundle collection is already in process. To cancel it, run:
```

micheleRP commented on src/go/rpk/pkg/cli/debug/remotebundle/start.go:
```diff
@@ -0,0 +1,217 @@
+			}
+			if anyOK {
+				fmt.Printf(`
+The debug bundle collection process has started with Job-ID %v, to check the 
+status, run:
```
```suggestion
The debug bundle collection process has started with Job-ID %v. To check the 
status, run:
```

gene-redpanda reviewed (dismissed):
LGTM

dotnwat reviewed (commented):
this commit

```
commit aa3b2d9b6abddf20aed875d93fabab6c5549e401 (HEAD)
Author: r-vasquez <rvasque3@gmail.com>
Date:   Fri Nov 1 10:07:37 2024 -0700

    rpk debug: move shared code to common

    These are options that are suitable for sharing
    with Debug Remote Bundle
```

appears to not be bisectable

```
ERROR: no such package 'src/go/rpk/pkg/cli/debug/common': BUILD file not found in any of the following directories. Add a BUILD file to a directory to mark it as a package.
 - /home/nwatkins/src/redpanda-bisect-check/redpanda/src/go/rpk/pkg/cli/debug/common
ERROR: /home/nwatkins/src/redpanda-bisect-check/redpanda/src/go/rpk/pkg/cli/debug/bundle/BUILD:3:11: no such package 'src/go/rpk/pkg/cli/debug/common': BUILD file not found in any of the following directories. Add a BUILD file to a directory to mark it as a package.
 - /home/nwatkins/src/redpanda-bisect-check/redpanda/src/go/rpk/pkg/cli/debug/common and referenced by '//src/go/rpk/pkg/cli/debug/bundle:bundle'
DEBUG: /home/nwatkins/.cache/bazel/_bazel_nwatkins/27f5827f173f731481084ebbdf273b17/external/rules_pkg~/pkg/mappings.bzl:310:34: same source mapped to different locations <generated file src/v/redpanda/redpanda> redpanda redpanda.runfiles/_main/src/v/redpanda/redpanda
WARNING: errors encountered while analyzing target '//src/go/rpk/cmd/rpk:rpk', it will not be built.
Analysis failed
WARNING: errors encountered while analyzing target '//src/go/rpk/pkg/cli/debug:debug', it will not be built.
Analysis failed
WARNING: errors encountered while analyzing target '//src/go/rpk/pkg/cli:cli', it will not be built.
Analysis failed
WARNING: errors encountered while analyzing target '//src/go/rpk/pkg/cli:cli_test', it will not be built.
Analysis failed
WARNING: errors encountered while analyzing target '//src/go/rpk/pkg/cli/debug/bundle:bundle', it will not be built.
Analysis failed
WARNING: errors encountered while analyzing target '//src/go/rpk/cmd/rpk:rpk_lib', it will not be built.
Analysis failed
WARNING: errors encountered while analyzing target '//src/go/rpk/pkg/cli/debug/bundle:bundle_test', it will not be built.
Analysis failed
INFO: Analyzed 1426 targets (26 packages loaded, 1094 targets configured).
WARNING: errors encountered while analyzing target '//:rpk', it will not be built.
Analysis failed
```

r-vasquez commented:
@dotnwat Thanks! good catch. Indeed I added the Bazel changes to the last commit. Fixed :+1:

gene-redpanda commented on src/go/rpk/pkg/cli/debug/bundle/bundle.go:
```diff
@@ -59,20 +60,8 @@ func NewCommand(fs afero.Fs, p *config.Params) *cobra.Command {
-		metricsSampleCount int
-		cpuProfilerWait    time.Duration
+		timeout   time.Duration
+		opts      common.DebugBundleSharedOptions
```
I like this model, it would definitely make sense for topic stuff as well.

michael-redpanda reviewed (dismissed):
In your ducktape tests, for all `asserts`, please provide a human readable error message

r-vasquez commented on tests/rptest/tests/rpk_debug_bundle_test.py:
```diff
@@ -0,0 +1,189 @@
+                filtered_errors.append(l)
+
+        assert not filtered_errors
+        assert output_file == file_path
```
Fixed, thanks!

michael-redpanda replied:
They look great!

r-vasquez commented:
[Force Push:](https://github.com/redpanda-data/redpanda/compare/b053c72b9ea427ed91cd0346aac0f38b03b2d2b2..afe8a82e02bdf80b95142ce31e047746db0ef73f)

- Add `assert` failure messages

r-vasquez commented:
/ci-repeat 3
skip-units
dt-repeat=20
tests/rptest/tests/rpk_debug_bundle_test.py::RpkDebugBundleTest

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/58586-->
<!--d26a6bf9-45c8-4106-bac9-f62961278a54-->
<!--title-->
## Retry command for [Build#58586](https://buildkite.com/redpanda/redpanda/builds/58586)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/rpk_debug_bundle_test.py::RpkDebugBundleTest.test_debug_bundle
tests/rptest/tests/rpk_debug_bundle_test.py::RpkDebugBundleTest.test_debug_bundle
tests/rptest/tests/rpk_debug_bundle_test.py::RpkDebugBundleTest.test_debug_bundle
tests/rptest/tests/rpk_debug_bundle_test.py::RpkDebugBundleTest.test_debug_bundle
tests/rptest/tests/rpk_debug_bundle_test.py::RpkDebugBundleTest.test_debug_bundle
tests/rptest/tests/rpk_debug_bundle_test.py::RpkDebugBundleTest.test_debug_bundle
tests/rptest/tests/rpk_debug_bundle_test.py::RpkDebugBundleTest.test_debug_bundle
tests/rptest/tests/rpk_debug_bundle_test.py::RpkDebugBundleTest.test_debug_bundle
tests/rptest/tests/rpk_debug_bundle_test.py::RpkDebugBundleTest.test_debug_bundle
tests/rptest/tests/rpk_debug_bundle_test.py::RpkDebugBundleTest.test_debug_bundle
tests/rptest/tests/rpk_debug_bundle_test.py::RpkDebugBundleTest.test_debug_bundle
tests/rptest/tests/rpk_debug_bundle_test.py::RpkDebugBundleTest.test_debug_bundle
tests/rptest/tests/rpk_debug_bundle_test.py::RpkDebugBundleTest.test_debug_bundle
tests/rptest/tests/rpk_debug_bundle_test.py::RpkDebugBundleTest.test_debug_bundle
```
<!--/body-->

r-vasquez commented:
/ci-repeat 3
skip-units
dt-repeat=20
tests/rptest/tests/rpk_debug_bundle_test.py::RpkDebugBundleTest.test_remote_debug_bundle_default

r-vasquez commented:
Force Push

- I found an issue in the test that we moved from `cluster_test.py` to this new file: changed to a random string to generate the name of the bundle to avoid possible races.
- Rebased with Dev to avoid the `build-bazel` error.

Note:

The new test added had a successful 60x run: https://buildkite.com/redpanda/redpanda/builds/58607#_, I'm going to retry aa 60x run again after the normal CI passes.

r-vasquez commented:
/ci-repeat 3
skip-units
dt-repeat=20
tests/rptest/tests/rpk_debug_bundle_test.py::RpkDebugBundleTest
